service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/ObservationMapper.java

```diff
@@ -116,29 +123,51 @@ private String mapObservationToNarrativeStatement(IdMapper idMapper, Observation
 
-        CodeableConceptMappingUtils.extractTextOrCoding(observation.getDataAbsentReason()).ifPresent(dataAbsentReason ->
+        CodeableConceptMappingUtils.extractTextOrCoding(observation.getDataAbsentReason()).ifPresent(dataAbsentReason -> {
+            String comment = DATA_ABSENT_PREFIX + dataAbsentReason;
+
             narrativeStatementsBlock.append(
-                prepareNarrativeStatement(observation, dataAbsentReason, CommentType.LABORATORY_RESULT_DETAIL.getCode(), idMapper)
-            )
-        );
+                prepareNarrativeStatement(observation, comment, CommentType.LABORATORY_RESULT_DETAIL.getCode(), idMapper)
+            );
+        });
 
         CodeableConceptMappingUtils.extractTextOrCoding(observation.getInterpretation()).ifPresent(interpretation ->
             narrativeStatementsBlock.append(
                 prepareNarrativeStatement(observation, interpretation, CommentType.LABORATORY_RESULT_DETAIL.getCode(), idMapper)
             )
         );
 
-        CodeableConceptMappingUtils.extractTextOrCoding(observation.getBodySite()).ifPresent(bodySite ->
+        CodeableConceptMappingUtils.extractTextOrCoding(observation.getBodySite()).ifPresent(bodySite -> {
+            String comment = BODY_SITE_PREFIX + bodySite;
+
             narrativeStatementsBlock.append(
-                prepareNarrativeStatement(observation, bodySite, CommentType.LABORATORY_RESULT_DETAIL.getCode(), idMapper)
-            )
-        );
+                prepareNarrativeStatement(observation, comment, CommentType.LABORATORY_RESULT_DETAIL.getCode(), idMapper)
+            );
+        });
+
+        CodeableConceptMappingUtils.extractTextOrCoding(observation.getMethod()).ifPresent(method -> {
+            String comment = METHOD_PREFIX + method;
 
-        CodeableConceptMappingUtils.extractTextOrCoding(observation.getMethod()).ifPresent(method ->
             narrativeStatementsBlock.append(
-                prepareNarrativeStatement(observation, method, CommentType.LABORATORY_RESULT_DETAIL.getCode(), idMapper)
-            )
-        );
+                    prepareNarrativeStatement(observation, comment, CommentType.LABORATORY_RESULT_DETAIL.getCode(), idMapper)
+            );
+        });
+
+        if (observation.hasReferenceRange()) {
+            Observation.ObservationReferenceRangeComponent referenceRange = observation.getReferenceRangeFirstRep();
+
+            if (observation.hasValueQuantity()) {
+                Optional<String> referenceRangeUnit = extractUnit(referenceRange);
+
+                if (referenceRangeUnit.isPresent() && isRangeUnitValid(referenceRangeUnit.get(), observation.getValueQuantity())) {
+                    String comment = RANGE_UNITS_PREFIX + referenceRangeUnit.get();
+
+                    narrativeStatementsBlock.append(
+                        prepareNarrativeStatement(observation, comment, CommentType.COMPLEX_REFERENCE_RANGE.getCode(), idMapper)
+                    );
+                }
```

Comment:
Functional-like version, if you prefer:
```suggestion
                extractUnit(referenceRange)
                    .filter(referenceRangeUnit -> isRangeUnitValid(referenceRangeUnit, observation.getValueQuantity()))
                    .map(RANGE_UNITS_PREFIX::concat)
                    .map(comment -> prepareNarrativeStatement(observation, comment, CommentType.COMPLEX_REFERENCE_RANGE.getCode(), idMapper))
                    .ifPresent(narrativeStatementsBlock::append);
```